### PR TITLE
Feat - 207- Profile Page Changes

### PR DESCRIPTION
**Pull request by maubarraganwize**
# What does this PR do?

- Modify the page to show profile information from our DB profile table. This will show information synced from datalake
- Show the projects the user is currently involved in
- Allow the user to specify its Github user name
- After saving the user name, retrieve the user info and repos from Github and populate the corresponding tables.

UPDATE:
- Only allow profile owner and Admins to edit Github username
- Made projects clickable
- Added link in profiles list to profile detail page

# Where should the reviewer start?

app/routes/profile/$email/index.tsx

# How should this be manually tested?

Log in and go to the Profile page
Edit the github username and save

# Any background context you want to provide?

# What are the relevant tickets?

[Ticket 207](https://github.com/wizeline/remix-project-lab/issues/207)

# Screenshots

Before: Without Github data:
image

Before: With Github data:
image

After:
image

# Questions

# Checklist

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

File: app/models/profile.server.ts [hunk 1]
```diff
@@ -182,26 +247,6 @@ export async function consolidateProfilesByEmail(
...
-//   updatedAt: string
-//   country: string
-//   location: string
-//   preferredName: string
-//   benchStatus: string
-//   businessUnit: string
-//   employeeStatus: string
-//   githubUser: string
-// }
-
```
Comment (martinrobled0) [inline app/models/profile.server.ts:204]: Why did you keep this ?

File: app/routes/profile/$email/index.tsx [hunk 2]
```diff
@@ -114,67 +171,204 @@ export const ProfileInfo = () => {
...
+                          ? profileData.githubUser
+                          : "<Not specified>"}
+                      </Typography>
+                      <IconButton
+                        type="button"
+                        onClick={() => {
+                          setIsEditGithubUserActive(true);
+                        }}
+                      >
+                        <EditSharp />
```
Comment (martinrobled0) [inline app/routes/profile/$email/index.tsx:311]: maybe we should only allow edit to admins and themselves

File: app/routes/profile/$email/index.tsx [hunk 3]
```diff
@@ -114,67 +171,204 @@ export const ProfileInfo = () => {
...
               <Paper elevation={0} sx={{ padding: 2 }}>
                 <h2 style={{ marginTop: 0, paddingLeft: 20 }}>
-                  Active Projects
+                  Github Active Projects
                 </h2>
                 <Grid container sx={{ p: 2 }}>
-                  {githubProjectsLink.map((project) => (
+                  {githubProjects.map((project) => (
                     <Grid item xs={12} key={project.id}>
                       <Card
```
Comment (martinrobled0) [inline app/routes/profile/$email/index.tsx:373]: also could be nice that the card projects will be clickable

**Comment by martinrobled0** [comment]
Maybe we could block some parts of the ui when the status it's terminated
image

**Comment by martinrobled0** [comment]
> Maybe we could block some parts of the ui when the status it's terminated image

and in this case I clicked the edit button and then clicked save and the page was redirected to my profile.
image

**Review by martinrobled0 — CHANGES_REQUESTED** [review]
I added some comments

File: app/models/profile.server.ts (hunk 1 above)
Comment (maubarraganwize) [reply app/models/profile.server.ts:204]: I didn't keep those, they are marked as removed 🤷 😄

Comment (martinrobled0) [reply app/models/profile.server.ts:204]: jeje it's true 😛

**Review by CarlosMilano — COMMENTED** [review]
the same as martin

**Comment by martinrobled0** [comment]
Now if I try to edit someone else profile I get this 
image

**Comment by martinrobled0** [comment]
> Now if I try to edit someone else profile I get this image

I fact, Is it okay that I can edit other people's profiles and comment the experience part?

File: app/routes/profile/$email/index.tsx (hunk 3 above)
Comment (maubarraganwize) [reply app/routes/profile/$email/index.tsx:426]: Fixed!

File: app/routes/profile/$email/index.tsx (hunk 2 above)
Comment (maubarraganwize) [reply app/routes/profile/$email/index.tsx]: Fixed!

**Review by martinrobled0 — APPROVED** [review]
LGTM

**Pull request merged**